### ISSUE TITLE
Error when using .load

### ISSUE THREAD

**Issue by fakerybakery** [gradio-app/gradio#8470]
### Describe the bug

This happens on the [TTS Arena](https://huggingface.co/spaces/TTS-AGI/TTS-Arena) when the Gradio version is bumped from 0.31 -> 0.32

```
leaderboard.load(get_leaderboard, inputs=[reveal_prelim], outputs=[df])
```

seems to trigger the `synthandreturn` function

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

https://huggingface.co/spaces/TTS-AGI/TTS-Arena/blob/main/app.py

### System Info

```shell
I was using the latest version of Gradio
```

### Severity

I can work around it

**Comment by abidlabs** [comment]
Just to confirm, you mean when gradio is bumped from `4.31` to `4.32`? And this issue is present even on `4.33` (current latest)?

**Comment by fakerybakery** [comment]
Yes, I was on the latest version before. I rolled back to 4.19 but tested it with 4.31 (worked) and 4.32 (didn't work)

**Comment by pngwn** [comment]
Will take a look. Could be do with some recent changes to the client.

**Comment by pngwn** [comment]
There is a lot going on in that demo, could you provide a minimal reproduction with no dependencies please.

**Comment by fakerybakery** [comment]
Hi, I upgraded from 4.36.0 to 4.36.1 and (strangely) it seemed to work again.

**Comment by fakerybakery** [comment]
Seems to have been documented in #8425 and fixed in #8504

**Comment by pngwn** [comment]
Awesome, thanks for checking!
